mqtt/mqtt-broker/src/auth.rs

```diff
@@ -0,0 +1,85 @@
+#![allow(dead_code)] // TODO @dmolokanov remove when implemented
+use std::fmt;
+
+use failure::Fail;
+
+use crate::Error;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum AuthId {
```

Comment:
I know most of the things here are self-explanatory, but maybe still add some code docs in this file? for example we can explain what is AuthId, how it's different from any other identities, etc...